pyo3-macros-backend/src/method.rs

```diff
@@ -178,6 +178,7 @@ impl SelfType {
                             .map_err(::std::convert::Into::<_pyo3::PyErr>::into)
                             .and_then(
                                 #[allow(clippy::useless_conversion)]  // In case slf is PyCell<Self>
+                                #[allow(unknown_lints, clippy::unnecessary_fallible_conversions)]  // In case slf is Py<Self>
```

Comment:
The `unknown_lints` is necessary for this to work on stable, right?

I am not sure we should commit to doing this until the lint hits stable as lints often change/are demoted before being released to stable.

Reply:
Also for any older release, I assume

Reply:
I wonder if we can also silence this lint by moving away from `TryFrom` to some other implementation, given that's what clippy is complaining about.

Reply:
I _assume_ (but didn't check in depth) that `TryFrom` is needed for other receiver types.

Reply:
It is, but we could replace it all with some other trait or function call

Reply:
In the end in #3587 we stuck with `From`, so I think it's correct to stick to `TryFrom`